packages/flutter_tools/lib/src/commands/run_mojo.dart

```diff
@@ -108,17 +107,31 @@ class RunMojoCommand extends FlutterCommand {
 
     if (argResults['android']) {
       args.add('--android');
-      final String appName = path.basename(appPath);
-      final String appDir = path.dirname(appPath);
-      args.add('mojo:launcher http://app/$appName');
-      args.add('--map-origin=http://app/=$appDir');
+    }
 
+    try {
+      final String appPath = _makePathAbsolute(bundlePath);
+      if (argResults['android']) {
+	final String appName = path.basename(appPath);
+	final String appDir = path.dirname(appPath);
+	args.add('mojo:launcher http://app/$appName');
+	args.add('--map-origin=http://app/=$appDir');
+      } else {
+        args.add('mojo:launcher file://$appPath');
+      }
+    } catch(_) {
```

Comment:
This catches too much stuff and could hide real error.  Can we not check for this explicitly?  For example, we could see if `bundlePath` starts with a valid URI scheme?  For example, we could look at the characters before the first `:` and see if they're all valid for a scheme https://tools.ietf.org/html/rfc3986#section-3.1:

```
scheme      = ALPHA *( ALPHA / DIGIT / "+" / "-" / "." )
```